pkg/controllers/webhookauthenticator/webhookauthenticator_controller.go

```diff
@@ -176,3 +164,57 @@ func (c *webhookAuthenticatorController) ensureKubeConfigSecret(ctx context.Cont
 	secret, _, err := resourceapply.ApplySecret(c.secrets, recorder, requiredSecret)
 	return secret, err
 }
+
+func (c *webhookAuthenticatorController) getAuthenticatorCertKeyPair() (key, cert []byte, err error) {
+	var waitingForCertKey *string
+	defer func() {
+		if waitingForCertKey != nil {
+			cond := operatorv1.OperatorCondition{
+				Type:    "AuthenticatorCertKeyProgressing",
+				Status:  operatorv1.ConditionFalse,
+				Reason:  "AsExpected",
+				Message: "All is well",
+			}
+
+			if len(*waitingForCertKey) > 0 {
```

Comment:
this three-state logic is hard to read

Reply:
I added an explanation to the variable, should make this easy to understand